src/README.md

```diff
@@ -89,31 +92,61 @@ Like gRPC, RPC are serialized via protocol buffers.
 [CART](https://github.com/daos-stack/cart) is a userspace function shipping library that provides low-latency high-bandwidth communications for the DAOS data plane. It supports RDMA capabilities and scalable collective operations. CART is built over [Mercury](https://github.com/mercury-hpc/mercury) and [libfabric](https://ofiwg.github.io/libfabric/). The CART library is used for all communications between libdaos and daos_io_server instances.
 
 <a id="3"></a>
-## DAOS Layering and Microservices
+## DAOS Layering and Services
 
 <a id="31"></a>
 ### Architecture
 
-The DAOS stack is organized as a set of microservices over a client/server architecture. The <a href="#5a">figure</a> below shows the logical layering of the DAOS stack.
+As shown in the diagram below, the DAOS stack is structured as a collection of storage services over a client/server architecture. Examples of DAOS services are the pool, container, object and rebuild services.
+
+![DAOS Internal Services & Libraries](/doc/graph/services.png)
+
+A DAOS service can be spread across the control and data planes and communicate internally through dRPC. Most services have a client and server components that can synchronize through gRPC or CART. Cross-service communications are always done through direct API calls. Those function calls can be invoked across either the client or server component of the services. While each DAOS service is designed to be fairly autonomous and isolated, some are more tightly coupled than others. That is typically the case of the rebuild service that needs to interact closely with the pool, container and object services to restore data redundancy after a DAOS server failure.
```

Comment:
spelling - "Most services have a client and server components" -> "Most services have client and server components"